src/pages/workspaces/workspace/applications/AppLauncher.js

```diff
@@ -55,6 +63,89 @@ const ApplicationLauncher = _.flow(
   const runtime = getCurrentRuntime(runtimes)
   const runtimeStatus = getConvertedRuntimeStatus(runtime) // preserve null vs undefined
 
+  const FileOutdatedModal = ({ onDismiss, bucketName }) => {
+    const handleChoice = _.flow(
+      withErrorReportingInModal('Error setting up analysis file syncing')(onDismiss),
+      Utils.withBusyState(setBusy)
+    )(async shouldCopy => {
+      // this modal only opens when the state variable outdatedAnalyses is non empty (keeps track of a user's outdated RStudio files). it gives users two options when their files are in use by another user
+      // 1) make copies of those files and continue working on the copies or 2) do nothing.
+      // in either case, their original version of the analysis is outdated and we will no longer sync that file to the workspace bucket for the current user
+      await Promise.all(_.flatMap(async ({ name, metadata: currentMetadata }) => {
+        const file = getFileName(name)
+        const newMetadata = currentMetadata
```

Comment:
Looks like we can eliminate one more `const`?

```suggestion
      await Promise.all(_.flatMap(async ({ name, metadata: newMetadata }) => {
        const file = getFileName(name)
```